### DISCUSSION TITLE
Next.js App hangs when deployed to Vercel

### DISCUSSION THREAD

**Discussion by ephraimduncan** [vercel/next.js#43484] [Help]
The version is Next 12.3

Everything is working locally and I don't get any deployment errors. The page hangs and never loads.

This is from Vercel. 
![Screenshot 2022-11-28 at 23 13 45](https://user-images.githubusercontent.com/55143799/204400439-fd0ddc4a-b2f9-4964-8e2c-36808f22e3ec.png)

No Screenshot Available. 
Vercel is not able to load the page itself

**Comment by icyJoseph** [comment]
What node version is running on Vercel?

> The page hangs and never loads.

What do you mean more specifically? I can see the page live, but I reckon the new deployment is failing?

**Reply by ephraimduncan** [reply]
I reversed the PR that caused it that is why you were able to see it. I figured it out. 
I was switching React with Preact for smaller buid size during production build and I decided to create dynamic OG Images using `@vercel/og`. 

`@vercel/og` doesn't support Preact so that was causing the hang and everthing so I had to remove Preact during builds and it is working as expected. 

This was the exact build url before I reversed everything. https://astrosaurus-4z88re5lv-dephraiim.vercel.app/

**Reply by icyJoseph** [reply]
Aha yeah that makes sense. Second time I see someone do this switch and run into issues. Styled components also doesn't seem to like Preact swaps. Good that you figure it out!
